tests/sos_tests.py

```diff
@@ -0,0 +1,464 @@
+        :type content:  ``str``
+        """
+        assert not re.match(".*%s.*" % content, self.cmd_output.stdout), "String '%s' present in stdout" % content
+
+    def assertPluginIncluded(self, plugin):
+        """Ensure that the specified plugin did run for the sos execution
+
+        Note that this relies on manifest.json being successfully created
+
+        :param plugin:  The name of the plugin
+        :type plugin:  `` str``
+        """
+        if not self.manifest:
+            self.error("No manifest found, cannot check for %s execution" % plugin)
+        assert plugin in self.manifest['components']['report']['plugins'].keys(), 'Plugin not recorded in manifest'
+
+    def assertPluginNotIncluded(self, plugin):
+        """Ensure that the specified plugin did NOT run for the sos execution
+        Note that this relies on manifest.json being successfully created
+
+        :param plugin:  The name of the plugin
+        :type plugin:  `` str``
+        """
+        if not self.manifest:
+            self.error("No manifest found, cannot check for %s execution" % plugin)
+        assert plugin not in self.manifest['components']['report']['plugins'].keys(), 'Plugin is recorded in manifest'
+
+    def assertOnlyPluginsIncluded(self, plugins):
+        """Ensure that only the specified plugins are in the manifest
+
+        :param plugins:  The plugin names
+        :type plugins:  ``str`` or ``list`` of strings
+        """
+        if not self.manifest:
+            self.error("No manifest found, cannot check for %s execution" % plugins)
+        if isinstance(plugins, str):
+            plugins = [plugins]
+        _executed = self.manifest['components']['report']['plugins'].keys()
+
+        # test that all requested plugins did run
+        for i in plugins:
+            assert i in _executed, "Requested plugin '%s' did not run" % i
+
+        # test that no unrequested plugins ran
+        for j in _executed:
+            assert j in plugins, "Unrequested plugin '%s' ran as well" % j
+
+class StageOneReportTest(BaseSoSReportTest):
+    """This is the test class to subclass for all Stage One (no mocking) tests
+    within the sos test suite.
+
+    In addition to any test_* methods defined in the test cases that subclass
+    this, the methods defined here will ALSO run, to ensure basic consistency
+    across test cases
+
+    NOTE: You MUST replace the following line in the docstring of your own
+    test cases, as otherwise the test will be disabled. This line is here to
+    prevent this base class from being treated as a valid test case. Also, if
+    you add any tests to this base class, make sure to add a line such as
+    ':avocado: tags=stageone' to ensure the base tests run with new test cases
+
+    :avocado: disable
+    """
+
+    sos_cmd = ''
+
+    def test_archive_created(self):
+        """Ensure that the archive tarball was created and has the right owner
+
+        :avocado: tags=stageone
```

Comment:
Is it expected that all classes that inherit from this will be `stageone`? If so, it should be possible to move the tags definition to the class' docstring (and avoid repetition).

Reply:
Ack. This is likely a hold over from when I was first playing with tags and trying to get these tests to run in anything that subclasses `StageOneReportTest`, but not run `StageOneReportTest` directly. I.E. before I found `:avocado: disable:` for the base class.